openedx/core/djangoapps/credit/signals.py

```diff
@@ -70,3 +82,209 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
                     api.set_credit_requirement_status(
                         username, course_id, 'grade', 'grade', status="failed", reason={}
                     )
+
+
+def tag_course_content_with_partition_scheme(course_key, partition_scheme):
+    """ Create user partitions with provided partition scheme and tag credit
+    blocks for the given course with those user partitions.
+
+    Args:
+        course_key (CourseKey): Identifier for the course
+        partition_scheme (str): Name of the user partition scheme
+
+    """
+    # TODO: Refactor code (move code to proper files e.g., 'common/lib/xmodule/xmodule/partitions/partitions.py') in once ICRV access control functionality is complete
+
+    # TODO: Dynamically update user_id (actual course publisher)
+    user_id = 1
+
+    # TODO: Increase logging (add logging for errors and success)
+    # get user partition with provided partition scheme
+    user_partition = UserPartition.get_scheme(partition_scheme)
+    if user_partition is None:
+        # log and raise exception 'NoSuchUserPartitionError' if no matching
+        # user partition exists
+        log.error(
+            u'No user partition found with scheme "%s". Exiting tagging course content with user partition.',
+            partition_scheme
+        )
+        raise NoSuchUserPartitionError
+
+    access_control_credit_xblocks = _get_credit_xblocks_for_access_control(course_key)
+    new_user_partitions = []
+    for block in access_control_credit_xblocks:
```

Comment:
We should probably wrap all of the write operations in a `bulk_operations` context manager.

It looks like we're lucky!  If I'm reading this correctly, the [bulk operations](https://github.com/edx/edx-platform/blob/756008aa9bd03ecd147ae98271a16fb640d59945/common/lib/xmodule/xmodule/modulestore/__init__.py#L174) implementation can be called with `emit_signals=False` to suppress firing the course published signal.

It would look something like:
```
with modulestore().bulk_operations(course_key, emit_signals=False):
   ...
```